src/error.c

```diff
@@ -208,6 +211,10 @@ static Obj FuncPRINT_CURRENT_STATEMENT(Obj self, Obj stream, Obj context)
 
             Int type = TNUM_STAT(call);
             Obj filename = GET_FILENAME_BODY(body);
+            if (!filename) {
+                filename = MakeImmString("<no filename>");
+            }
```

Comment:
But the line the also will be bogus. I would suggest to instead change the `else {` above to `else if (filename != 0) {` (and then of course declare and init `filename` earlier)